### ISSUE TITLE
[ADK] Improve AOE Recognition

### ISSUE THREAD

**Issue by geraint425** [Hekili/hekili#232]
Hello. I would like to talk to the swami about the automatic recognition of AOE. I've noticed that if I set the goals for the plague. Then in rotation I am immediately offered death and decay, although the goals are very far away and not in my melee radius. 
Is it possible to somehow do so that at such moments in the rotation death and decay are not offered if the goals are far away. And just an epidemic was proposed because it does not depend on what the distance is.

**Comment by Hekili** [comment]
Post your actual settings.  If you want multi-target recommendations to be based on range, you use range-based target detection.

**Comment by geraint425** [comment]
https://imgur.com/XjmcFc6 My Settings. https://imgur.com/daejjD7 
And he attached a clear screenshot where he showed how much the plague hangs on targets. marked in red. I also noted in red the rotation shown at the moment that the addon suggested to me. And he circled with a circle of ornamental circles the radius that I can hook on them. In the right way. The addon should not offer me to use death and decay, but simply offer to push the epidemic. 
If the goals were close to me and within the range of my abilities then yes the addon would have to offer me correctly. But we play for a class such that we have a plague, and we have an epidemic that works at any distance, it’s enough just to have our plague for purposes so that our epidemic works. Therefore, I asked if it is possible that such classes that are dependent on AOE at far and near distances work correctly? Because at the same time the radius of targets and dots does not work.

**Comment by Hekili** [comment]
So, the settings, for clarity's sake:

Use Nameplate Detection - counts targets within a certain range, specified by you, and requires enemy nameplates to be active.

Detect Damaged Enemies - counts enemies that have been hit by you, regardless of range (i.e., you could've hit something when it was in melee range and then it wanders away).

Detect Dotted Enemies - counts enemies that have your debuffs on them, regardless of range (same issue).

Filter Damaged Enemies by Range - applies a filter to damaged/dotted enemies based on range, if we recently got their range information (so, basically, this requires nameplate detection to be active).

In short, you're not configuring the addon to make recommendations for what you want.  And I gather that you want the addon to count all dotted targets for Epidemic and only targets in melee range for Death and Decay.  That makes sense but is not presently supported.

**Comment by Hekili** [comment]
Also, bear in mind for Epidemic, there is no way for the addon to know if enemies are clustered EXCEPT if they are very close to you (i.e., melee).  You could have 4 dotted enemies, but each of them are 20 yards in a different direction from you and hitting Epidemic would do nothing.  The addon has no way of knowing (it is not available via the WoW API).

**Comment by geraint425** [comment]
And how is it then implemented in Hero Rotation? I’m already used to your addon and don’t really want to get used to theirs. But there it turns out that it is realized that in close combat death and decay work as soon as targets appear in our close radius. And until this time, while the targets are far away but have a plague, the addon offers to push the epidemic and if these goals have already reached us, then only death and decay will occur. They even realized what shows next that there are goals for which there is no plague, and if the goal remains less squabble, they suggest using the reaper to get an additional debuff. 

Or please show how it is possible to partially implement this if you can in the screenshots settings. I will be very grateful for your help.

**Comment by Hekili** [comment]
> And how is it then implemented in Hero Rotation?

I don't know, but you can use Hero Rotation.  If they are able to reliably tell positioning (i.e., clusters) for enemies that are not close by, that's something that Blizzard is likely to break.

I'll explore a bit regarding target counting based on different criteria for Epidemic vs. Death and Decay.

**Comment by geraint425** [comment]
So the fact of the matter is that I do not want to use their addon) 
I’ve been using yours for 2 years and as practice has shown, it’s better that it doesn’t exist. Just why I compared it with the addon that they have such buns that would be great in yours. I've attached a video how it works. about the epidemic. 
so you can clearly see how it works for them. I really want you to work the same way. https://drive.google.com/file/d/1WDhRMmRYXaxy98ZMeVRFCv1G3QggPKrG/view?usp=sharing

**Comment by geraint425** [comment]
Okay, thanks, I close the topic, I realized that you can’t implement this functionality. It is unfortunate that you cannot.
